articles/content_filtering.md

```diff
@@ -0,0 +1,308 @@
+
+## ContentFilteredTopic
+
+ContentFilteredTopic describes a more sophisticated subscription that indicates the subscriber does not want to necessarily see all values of each instance published under the Topic.
+Rather, it wants to see only the values whose contents satisfy certain criteria.
+This class therefore can be used to request content-based subscriptions. 
+
+### Specification
+
+- Multiple ContentFilteredTopicDataReader exist and filtering is dependent on each ContentFilteredTopicDataReader's logical filtering expression and parameters.
+- Subscription is responsible to create ContentFilteredTopic.
+- ContentFilteredTopicDataWriter does filtering for each ContentFilteredTopicDataReader endpoint.
+  Basically it is likely that ContentFilteredTopicDataWriter does filtering on writer side, so that it can provide the improvement and efficiency especially for network resource.
+  But it is dependent on implementation if always ContentFilteredTopicDataWriter does filtering.
+  This is because if there is a lot of ContentFilteredTopicDataReader with complicated filtering expression and parameter, it would be probably more efficient just to send the messages to all ContentFilteredTopicDataReader. (for example, if the message type includes sequence, DataWriter needs to recreate the message for each DataReader based on filtering expression and parameters.)
+  And ContentFilteredTopicDataReader can filter out the message if not necessary based on filtering expression and parameter.
+  This is also said that ContentFilteredTopicDataReader does need to do filtering always just in case ContentFilteredTopicDataWriter ends up sending w/o filtering.
+  <img src="../img/content_filter/contents_filter_overview.png" width="600">
+- The filtering is done by means of evaluating a logical expression that involves the values some of the data-fields in the sample.
+  The logical expression is derived from the filter_expression and expression_parameters arguments.
+- Each ContentFilteredTopic has a single filter_expression and a sequence of expression_parameters.
+  ```
+  [parameter use cases]
+  "node = %0 OR node = %1" spinal_node attention_node : subscribe all parameter events from spinal and attention nodes
+  "node = %0 OR name = %1" audio_node attention_level : subscribe all parameter events from audio_node and parameter named attention_level of all nodes.
+  "(node = %0 AND name = %1) OR (node = %2 AND name = %3)" eye_node brightness ear_node loudness : subscribe brightness parameter of eye_node and loudness parameter of ear_node only.
+  [action use case]
+  "uuid = %0 OR uuid = %1" DEADBEEF CAFEFEED : subscribe only interest goal ids which client handlers possess.
+  ```
+- A single filter_expression and a sequence of expression_parameters can be modified dynamically at runtime.

```

Comment:
I think it would be good to have a full specification of what's a valid filter_expression, even if it's a link to the specification used by DDS.

Reply:
that is actually good idea.